### PR TITLE
Suppress additional output if --silent has been set.

### PR DESCRIPTION
**Pull request by cainesi**
This makes the script a bit closer to silent when "silent" operation has been specified.

### CODE REVIEW

File: cx_Freeze/freezer.py [hunk 1]
```diff
@@ -572,7 +572,9 @@ def _WriteModules(self, fileName, finder):
 
         if not self.silent:
             self._PrintReport(fileName, modules)
-        finder.ReportMissingModules()
+
+        if not self.silent:
```
Comment (marcelotduarte) [inline cx_Freeze/freezer.py]: An indentation is sufficient to place it in the silent block. But, I wonder why the original author did not do this.

**Review by marcelotduarte — CHANGES_REQUESTED** [review]

File: cx_Freeze/freezer.py (hunk 1 above)
Comment (cainesi) [reply cx_Freeze/freezer.py]: I know. I put it in a separate block as a hint to people in the future that they may want to consider different conditions for calling the ReportMissingModules() method.

I guess that the original author probably did not make the method conditional on "no self.self", because he viewed the missing modules as an error that the user should be informed about in any case. However, the problem is that ReportMissingModules() reports many "missing" modules no matter what-- for example, platform specific modules that are not available on other modules.

We could put in a safe list, of modules that we know are okay to be missing, and ignore those ones.  But that seems a lot more complicted.

Comment (marcelotduarte) [reply cx_Freeze/freezer.py]: > reports many "missing" modules no matter what

It also bothers me and I've tried to reduce it, because sometimes it helps to identify what is wrong, but with a very long list, it doesn't help at all ...

Comment (cainesi) [reply cx_Freeze/freezer.py]: Yeah. 

I think if someone knows enough to be helped by the list of missing modules, then they would also know enough to turn off "silent" when they encounter a problem.

File: cx_Freeze/macdist.py [hunk 2]
```diff
@@ -31,9 +31,13 @@ class bdist_dmg(Command):
     def initialize_options(self):
         self.volume_label = self.distribution.get_fullname()
         self.applications_shortcut = False
+        self.silent = None
```
Comment (marcelotduarte) [inline cx_Freeze/macdist.py:38]: Is this option documented for this class?

File: cx_Freeze/freezer.py (hunk 1 above)
Comment (cainesi) [reply cx_Freeze/freezer.py]: Incidentally, I was looking at distutils.  The way it handles the verbosity is by setting a global reporting threshold in the distutils.log module, and directing all the output through distutils.log.log() (and some sibling functions).  Should cx_Freeze standardize on doing that as well?

File: cx_Freeze/macdist.py [hunk 3]
```diff
@@ -31,9 +31,13 @@ class bdist_dmg(Command):
...
 
     def finalize_options(self):
-        pass
+        self.set_undefined_options(
+            "build_exe",
```
Comment (marcelotduarte) [inline cx_Freeze/macdist.py]: Why did you add the option build_exe? Can you also add documentation?

File: cx_Freeze/freezer.py (hunk 1 above)
Comment (cainesi) [reply cx_Freeze/freezer.py]: Are you okay having ReportMissingModules() silenced?  If you still want ReportMissingModules() to run by default, we could add another option specifically for it ("--hide-missing"?).

**Review by marcelotduarte — CHANGES_REQUESTED** [review]
I was a little reluctant about this change. I think you can leave it as it was and in another patch implement this new option.

File: cx_Freeze/macdist.py (hunk 2 above)
Comment (cainesi) [reply cx_Freeze/macdist.py:38]: My thinking was that bdist_dmg would just pick up the behavior specified for build_exe (which already has a silent option), without really defining a separate option for bdist_dmg.  But if you'd prefer I explicitly define and document a separate option for bdist_dmg, I am happy to do that instead.

Comment (marcelotduarte) [reply cx_Freeze/macdist.py:38]: Please, document it.

Comment (cainesi) [reply cx_Freeze/macdist.py:38]: > I was a little reluctant about this change. I think you can leave it as it was and in another patch implement this new option.

Done.  There is now a no_missing_warnings option on build_exe to suppress the list of missing modules.

One thought, for the future--I think if we replace the print & shutil.copyfile/copytree code with calls to distutils.dir_util.copy_tree and distutils.file_util.copy_file, we could piggy back off the distutils logging code and the distutils verbosity level would also affect cx_freeze's output.  Too busy to do that at the moment.

File: cx_Freeze/macdist.py (hunk 3 above)
Comment (cainesi) [reply cx_Freeze/macdist.py]: Fixed.

**Comment by cainesi** [comment]
> I was a little reluctant about this change. I think you can leave it as it was and in another patch implement this new option.

Sorry-- I misunderstood this request at first.  I've now separated out the "no_missing_warnings" into a separate pull request (#883).  This PR now only adds a silent flag for bdist_dmg and suppresses some file copying messages in build_exe.

**Comment by marcelotduarte** [comment]
@cainesi You removed the docs in doc/src/distutils.rst, can you add it?
image

**Comment by cainesi** [comment]
> @cainesi You removed the docs in doc/src/distutils.rst, can you add it?

Sorry that was a mistake.  I had only meant to remove the "no_missing_warnings" options, which is moved to a different pull request.  I have put the documentation of the bdist_dmg "silent" option back in.

**Comment by marcelotduarte** [comment]
Thanks!
